packages/language/src/preprocessor/instruction-interpreter.ts

```diff
@@ -1385,13 +1383,15 @@ function parseAndEvaluateProcedure(
       context,
     );
   }
-
   const localContext = createLocalContext(context, procedure);
   let value = runProcedure(procedure, evaluatedArgs, localContext);
-  if (isScalarValue(value) && suffix) {
+  if (isScalarValue(value) && immediateFollow && tokens[index + advance]) {
     // Add the suffix to the value, as it was immediately following the procedure call
     // Otherwise, the suffix will generate a separate token, which is not the intended behavior
-    value = stringToValue(value.value + suffix);
+    const nextToken = tokens[index + advance];
```

Comment:
The `index+advance` is a bit weird. It jumps directly after the invocation, or?
Maybe name the token: "nextTokenAfterProcedureInvocation".